*** a/cci-1.0.0/packages/rapidjson/include/rapidjson/document.h	2018-07-11 08:24:31.520031354 +0200
--- b/cci-1.0.0/packages/rapidjson/include/rapidjson/document.h	2018-07-11 08:51:58.559221920 +0200
***************
*** 1425,1431 ****
          MemberIterator pos = MemberBegin() + (first - MemberBegin());
          for (MemberIterator itr = pos; itr != last; ++itr)
              itr->~Member();
!         std::memmove(&*pos, &*last, static_cast<size_t>(MemberEnd() - last) * sizeof(Member));
          data_.o.size -= static_cast<SizeType>(last - first);
          return pos;
      }
--- 1425,1431 ----
          MemberIterator pos = MemberBegin() + (first - MemberBegin());
          for (MemberIterator itr = pos; itr != last; ++itr)
              itr->~Member();
!         std::memmove((void*)&*pos, &*last, static_cast<size_t>(MemberEnd() - last) * sizeof(Member));
          data_.o.size -= static_cast<SizeType>(last - first);
          return pos;
      }
***************
*** 1629,1635 ****
          ValueIterator pos = Begin() + (first - Begin());
          for (ValueIterator itr = pos; itr != last; ++itr)
              itr->~GenericValue();       
!         std::memmove(pos, last, static_cast<size_t>(End() - last) * sizeof(GenericValue));
          data_.a.size -= static_cast<SizeType>(last - first);
          return pos;
      }
--- 1629,1635 ----
          ValueIterator pos = Begin() + (first - Begin());
          for (ValueIterator itr = pos; itr != last; ++itr)
              itr->~GenericValue();       
!         std::memmove((void*)pos, last, static_cast<size_t>(End() - last) * sizeof(GenericValue));
          data_.a.size -= static_cast<SizeType>(last - first);
          return pos;
      }
***************
*** 1936,1942 ****
          if (count) {
              GenericValue* e = static_cast<GenericValue*>(allocator.Malloc(count * sizeof(GenericValue)));
              SetElementsPointer(e);
!             std::memcpy(e, values, count * sizeof(GenericValue));
          }
          else
              SetElementsPointer(0);
--- 1936,1942 ----
          if (count) {
              GenericValue* e = static_cast<GenericValue*>(allocator.Malloc(count * sizeof(GenericValue)));
              SetElementsPointer(e);
!             std::memcpy((void*)e, values, count * sizeof(GenericValue));
          }
          else
              SetElementsPointer(0);
***************
*** 1949,1955 ****
          if (count) {
              Member* m = static_cast<Member*>(allocator.Malloc(count * sizeof(Member)));
              SetMembersPointer(m);
!             std::memcpy(m, members, count * sizeof(Member));
          }
          else
              SetMembersPointer(0);
--- 1949,1955 ----
          if (count) {
              Member* m = static_cast<Member*>(allocator.Malloc(count * sizeof(Member)));
              SetMembersPointer(m);
!             std::memcpy((void*)m, members, count * sizeof(Member));
          }
          else
              SetMembersPointer(0);

