diff -rupN scv-2.0.0a-20161019/configure scv-2.0.0a-20161019-patched/configure
--- scv-2.0.0a-20161019/configure	2016-10-19 23:25:31.000000000 +0200
+++ scv-2.0.0a-20161019-patched/configure	2017-10-28 17:00:36.367027496 +0200
@@ -16136,7 +16136,7 @@ sparc)
 	case "$GXX" in
 	yes) echo 'setting compiler flags for g++'
 	     AM_CXXFLAGS="$SPARC_GNU_CXXFLAGS"
-	     CXXFLAGS="$GNU_OPT $GNU_DBG"
+	     CXXFLAGS="$CXXFLAGS $GNU_OPT $GNU_DBG"
 	     CXXFLAGS_DEBUG="$DEF_GNU_DBG"
 	     CXXFLAGS_OPT="$DEF_GNU_OPT"
 	     AM_CFLAGS="$SPARC_GNU_CFLAGS"
@@ -16152,8 +16152,8 @@ sparc)
 	     echo 'setting library options for g++ on Solaris'
 	     ;;
 	'')  echo 'setting compiler flags for NOT g++'
-	     	     	     	     	     	     	     AM_CXXFLAGS="$SPARC_CC_CXXFLAGS"
-	     CXXFLAGS="$SUN_CC_OPT $SUN_CC_DBG"
+	     AM_CXXFLAGS="$SPARC_CC_CXXFLAGS"
+	     CXXFLAGS="$CXXFLAGS $SUN_CC_OPT $SUN_CC_DBG"
 	     CXXFLAGS_DEBUG="$DEF_SUN_CC_DBG"
 	     CXXFLAGS_OPT="$DEF_SUN_CC_OPT"
 	     AM_CFLAGS="$SPARC_CC_CFLAGS"
@@ -16173,7 +16173,7 @@ hppa*)
 	case "$GXX" in
         yes) echo 'setting compiler flags for g++'
 	     AM_CXXFLAGS="$HPPA_GNU_CXXFLAGS"
-	     CXXFLAGS="$GNU_OPT $GNU_DBG"
+	     CXXFLAGS="$CXXFLAGS $GNU_OPT $GNU_DBG"
 	     CXXFLAGS_DEBUG="$DEF_GNU_DBG"
 	     CXXFLAGS_OPT="$DEF_GNU_OPT"
 	     AM_CFLAGS="$HPPA_GNU_CFLAGS"
@@ -16192,7 +16192,7 @@ hppa*)
 	     ;;
         '')  echo 'setting compiler flags and defines for HP aCC'
 	     AM_CXXFLAGS="$HPPA_ACC_CXXFLAGS"
-	     CXXFLAGS="$HP_ACC_OPT $HP_ACC_DBG"
+	     CXXFLAGS="$CXXFLAGS $HP_ACC_OPT $HP_ACC_DBG"
 	     CXXFLAGS_DEBUG="$DEF_HP_ACC_DBG"
 	     CXXFLAGS_OPT="$DEF_HP_ACC_OPT"
 	     AM_CFLAGS="$HPPA_ACC_XFLAGS"
@@ -16216,7 +16216,7 @@ powerpc*)
 	case "$GXX" in
         yes) echo 'setting compiler flags for g++'
 	     AM_CXXFLAGS="$LINUX_GNU_CXXFLAGS"
-	     CXXFLAGS="$GNU_OPT $GNU_DBG"
+	     CXXFLAGS="$CXXFLAGS $GNU_OPT $GNU_DBG"
 	     CXXFLAGS_DEBUG="$DEF_GNU_DBG"
 	     CXXFLAGS_OPT="$DEF_GNU_OPT"
 	     AM_CFLAGS="$LINUX_GNU_CFLAGS"
@@ -16235,7 +16235,7 @@ powerpc*)
 	     ;;
         '')  echo 'setting compiler flags for NOT g++'
 	     AM_CXXFLAGS="$LINUX_GNU_CXXFLAGS"
-	     CXXFLAGS="$GNU_OPT $GNU_DBG"
+	     CXXFLAGS="$CXXFLAGS $GNU_OPT $GNU_DBG"
 	     CXXFLAGS_DEBUG="$DEF_GNU_DBG"
 	     CXXFLAGS_OPT="$DEF_GNU_OPT"
 	     AM_CFLAGS="$LINUX_GNU_CFLAGS"
@@ -16256,7 +16256,7 @@ i[3456]86*)
 	case "$GXX" in
         yes) echo 'setting compiler flags for g++'
 	     AM_CXXFLAGS="$LINUX_GNU_CXXFLAGS"
-	     CXXFLAGS="$GNU_OPT $GNU_DBG"
+	     CXXFLAGS="$CXXFLAGS $GNU_OPT $GNU_DBG"
 	     CXXFLAGS_DEBUG="$DEF_GNU_DBG"
 	     CXXFLAGS_OPT="$DEF_GNU_OPT"
 	     AM_CFLAGS="$LINUX_GNU_CFLAGS"
@@ -16275,7 +16275,7 @@ i[3456]86*)
 	     ;;
         '')  echo 'setting compiler flags for NOT g++'
 	     AM_CXXFLAGS="$LINUX_GNU_CXXFLAGS"
-	     CXXFLAGS="$GNU_OPT $GNU_DBG"
+	     CXXFLAGS="$CXXFLAGS $GNU_OPT $GNU_DBG"
 	     CXXFLAGS_DEBUG="$DEF_GNU_DBG"
 	     CXXFLAGS_OPT="$DEF_GNU_OPT"
 	     AM_CFLAGS="$LINUX_GNU_CFLAGS"
@@ -16296,7 +16296,7 @@ x86_64*)
 	case "$GXX" in
         yes) echo 'setting compiler flags for g++'
 	     AM_CXXFLAGS="$LINUX_GNU_CXXFLAGS"
-	     CXXFLAGS="$GNU_OPT $GNU_DBG"
+	     CXXFLAGS="$CXXFLAGS $GNU_OPT $GNU_DBG"
 	     CXXFLAGS_DEBUG="$DEF_GNU_DBG"
 	     CXXFLAGS_OPT="$DEF_GNU_OPT"
 	     AM_CFLAGS="$LINUX_GNU_CFLAGS"
@@ -16315,7 +16315,7 @@ x86_64*)
 	     ;;
         '')  echo 'setting compiler flags for NOT g++'
 	     AM_CXXFLAGS="$LINUX_GNU_CXXFLAGS"
-	     CXXFLAGS="$GNU_OPT $GNU_DBG"
+	     CXXFLAGS="$CXXFLAGS $GNU_OPT $GNU_DBG"
 	     CXXFLAGS_DEBUG="$DEF_GNU_DBG"
 	     CXXFLAGS_OPT="$DEF_GNU_OPT"
 	     AM_CFLAGS="$LINUX_GNU_CFLAGS"
diff -rupN scv-2.0.0a-20161019/src/scv/_scv_introspection.h scv-2.0.0a-20161019-patched/src/scv/_scv_introspection.h
--- scv-2.0.0a-20161019/src/scv/_scv_introspection.h	2016-10-19 23:25:31.000000000 +0200
+++ scv-2.0.0a-20161019-patched/src/scv/_scv_introspection.h	2017-10-28 10:54:04.359018484 +0200
@@ -553,7 +553,7 @@ public: \
   _SCV_MAP(int,length) \
   _SCV_MAP(int,size) \
   sc_logic_value_t get_bit(int i) const \
-    { this->initialize(); return this->_get_instance()->get_bit(i); } \
+    { this->initialize(); return sc_logic_value_t(this->_get_instance()->get_bit(i)); } \
   void set_bit(int i, sc_logic_value_t v) \
     { this->initialize(); this->_get_instance()->set_bit(i,v); this->trigger_value_change_cb(); } \
   unsigned long get_word(int i) const \
diff -rupN scv-2.0.0a-20161019/src/scv/scv_random.cpp scv-2.0.0a-20161019-patched/src/scv/scv_random.cpp
--- scv-2.0.0a-20161019/src/scv/scv_random.cpp	2016-10-19 23:25:31.000000000 +0200
+++ scv-2.0.0a-20161019-patched/src/scv/scv_random.cpp	2017-10-28 10:56:49.880659476 +0200
@@ -49,6 +49,7 @@
 
 #include <cstdio>
 #include <cstdlib>
+#include <cstring>
 
 
 #if ((defined _MSC_VER) || (defined _WIN32))
diff -rupN scv-2.0.0a-20161019/src/scv/scv_util.cpp scv-2.0.0a-20161019-patched/src/scv/scv_util.cpp
--- scv-2.0.0a-20161019/src/scv/scv_util.cpp	2016-10-19 23:25:31.000000000 +0200
+++ scv-2.0.0a-20161019-patched/src/scv/scv_util.cpp	2017-10-28 10:58:02.323613714 +0200
@@ -44,7 +44,7 @@
 #include "scv/scv_util.h"
 #include "scv/_scv_associative_array.h"
 #include "scv/scv_report.h"
-
+#include <cstring>
 /* ************************************************************************** */
 
 // For cdsIdent:
